fdbserver/storageserver.actor.cpp

```diff
@@ -6194,11 +6194,17 @@ ACTOR Future<Void> fetchKeys(StorageServer* data, AddingShard* shard) {
 			// At iteration 1, dest SS selects GRV as fetchVersion and (suppose) can read the data from src SS.
 			// Then dest SS waits its version catch up with this GRV version and write the data to disk.
 			// Note that dest SS waits outside the fetchKeysParallelismLock.
+			fetchVersion = std::max(shard->fetchVersion, data->version.get());
 			if (lastError.code() == error_code_transaction_too_old) {
-				Version grvVersion = wait(tr.getRawReadVersion());
-				fetchVersion = std::max(grvVersion, data->version.get());
-			} else {
-				fetchVersion = std::max(shard->fetchVersion, data->version.get());
+				try {
+					// Note that error in getting GRV doesn't affect any storage server state. Therefore, we catch all
+					// errors here without failing the storage server. When error happens, fetchVersion fall back to
+					// the above computed fetchVersion.
+					Version grvVersion = wait(tr.getRawReadVersion());
+					fetchVersion = std::max(grvVersion, fetchVersion);
+				} catch (Error& e) {
+					TraceEvent(SevWarn, "FetchKeyGRVError", data->thisServerID).error(e);
```

Comment:
lastError = e?

Reply:
Added.

It seems any updates for `lastError` here will get overwritten below when reads return error.

Reply:
I suppose being overwritten is fine?

Reply:
Correct. That will be the real last error.